### PR TITLE
Add Websocket to process details page for live updates

### PR DESCRIPTION
**Pull request by tjeerddie**
- - [x] connect to websocket endpoint.
- - [x] Fallback to http call on websocket error.
- - [x] Authorization with access token.
- - [x] show form when necessary.
- - [x] reconnect silently with auth check.

### CODE REVIEW

**Comment by tjeerddie** [comment]
forms are shown when given in the step data but going back to the process tab it does not update and gives `Can't perform a React state update on an unmounted component` error.

File: src/pages/ProcessDetail.tsx [hunk 1]
```diff
@@ -79,47 +81,110 @@ class ProcessDetail extends React.PureComponent<IProps, IState> {
...
+    componentDidMount = () => {
+        const client = websocketService.connect(`api/processes/test/${this.props.match.params.id}`);
+        this.setState({ client: client });
+        client.onmessage = ({ data }) => {
+            if (typeof data === "string") {
+                try {
+                    const json = JSON.parse(data);
+                    const process = json.process;
+                    const step: Step = json.step;
+                    const error = json.error;
```
Comment (mrijk) [inline src/pages/ProcessDetail.tsx:159]: or something like:

`const {process, step, error} = JSON.parse(data);`

File: src/pages/ProcessDetail.tsx [hunk 2]
```diff
@@ -79,47 +81,110 @@ class ProcessDetail extends React.PureComponent<IProps, IState> {
...
+                        this.initializeProcessDetails(process);
+                    }
+                    if (step) {
+                        this.updateProcessStep(step);
+                    }
+                    if (error) {
+                        this.handleWebsocketError(error);
+                    }
+                } catch (e) {
+                    console.error("invalid JSON");
```
Comment (mrijk) [inline src/pages/ProcessDetail.tsx:170]: Haven't looked into detail, but don't we have a logger for that (that might eventually dump to the console, Sentry, both...)

File: src/websocketService/codes.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
+// Defines constants for all websocket close event codes that are used in practice
+// Link: https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent
+
+/**
+ * Normal closure; the connection successfully completed whatever purpose for which it was created.
+ */
+export const NORMAL_CLOSURE = 1000;
```
Comment (mrijk) [inline src/websocketService/codes.ts:7]: I can imagine that there is a module for that that predefines these constant, but haven't checked

Comment (tjeerddie) [reply src/websocketService/codes.ts:7]: I can't easily find any, this is what I found of someones gist
https://gist.github.com/RobertMenke/443e5cc57e4569c48b08c9592740da77

Comment (pboers1988) [reply src/websocketService/codes.ts:7]: 👍🏼

**Review by pboers1988 — APPROVED** [review]

File: src/pages/ProcessDetail.tsx (hunk 2 above)
Comment (pboers1988) [reply src/pages/ProcessDetail.tsx:185]: Sentry will automatically catch these exceptions. You could add a post to the user-error endpoint.

**Pull request merged**